enhancements/network/user-defined-network-segmentation.md

```diff
@@ -379,11 +382,283 @@ any user-defined primary network.
+    // +optional
+    ExcludeSubnets []string `json:"excludeSubnets,omitempty"`
+    
+    // Subnet used inside the OVN network topology.
+    // This field is ignored for non-primary networks (e.g.: Role Secondary).
+    // When omitted, this means no opinion and the platform is left to choose a reasonable default which is subject to change over time.    
+    // +kubebuilder:validation:XValidation:rule="1 <= size(self) && size(self) <= 2", message="Unexpected number of join subnets"
+    // +optional
+    JoinSubnets []string `json:"joinSubnets,omitempty"`
+    
+    // Control IP addresses management lifecycle.
+    // When `Persistent` is specified it enable workloads have persistent IP addresses.
+    // For example: Virtual Machines will have the same IP addresses along their lifecycle (stop, start migration, reboots).
+    // Supported by Topology `Layer2` and `Localnet`.
+    // +optional
+    IPAMLifecycle NetworkIPAMLifecycle `json:"ipamLifecycle,omitempty"`
+}
+```
+Suggested API validation rules:
+- `Topology` and `Role` fields are mandatory.
+- `Topology` can be one of `Layer2`, `Layer3`, `Localnet`.
+- `Role` can be one of `Primary`, `Secondary`.
+- `IPAMLifecycle` can be `Persistent`.
+- `JoinSubnets` length can be 1 or 2.
+
+##### Cluster scoped CRD
+```golang
+// ClusterUserDefinedNetwork describes shared OVN network across namespaces request.
+type ClusterUserDefinedNetwork struct {
+    metav1.TypeMeta   `json:",inline"`
+    metav1.ObjectMeta `json:"metadata,omitempty"`
+    // +kubebuilder:validation:Required
+    // +required
+    Spec ClusterUserDefinedNetworkSpec `json:"spec"`
+    // +optional
+    Status ClusterUserDefinedNetworkStatus `json:"status,omitempty"`
+}
+
+// ClusterUserDefinedNetwork defines the desired state of ClusterUserDefinedNetwork.
+type ClusterUserDefinedNetwork struct {
+    // NamespaceSelector Label selector for which namespace network should be available for.
+    // +kubebuilder:validation:Required
+    // +required
+    NamespaceSelector metav1.LabelSelector `json:"namespaceSelector"`
+    
+    // Template is direct specification of UserDefinedNetwork.
+    // +kubebuilder:validation:Required
+    // +required
+    Template *UserDefinedNetworkTemplateSpec `json:"template"`
+}
+
+// UserDefinedNetworkTemplateSpec UserDefinedNetwork spec template.
+type UserDefinedNetworkTemplateSpec struct {
+    // UserDefinedNetworkSpec contains the UserDefinedNetwork specification.
+    Spec UserDefinedNetworkSpec `json:"spec,omitempty"`
+}
+
+// ClusterUserDefinedNetworkStatus contains the observed status of the ClusterUserDefinedNetwork.
+type ClusterUserDefinedNetworkStatus struct {
+    // ActiveNamespaces indicates in which namespaces network is available.
+    ActiveNamespaces []string `json:"activeNamespaces,omitempty"`
+    
+    Conditions []metav1.Condition `json:"conditions,omitempty"`
+}
+```
+
+Suggested CRD short-name: `cudn`
```

Comment:
As above, short names are typically not suggested for low volume CRDs and reserved for commonly used APIs like core application resources